src/NHibernate.Test/NHSpecificTest/NH1553/MsSQL/SnapshotIsolationUpdateConflictTest.cs

```diff
@@ -134,22 +136,42 @@ protected override bool AppliesTo(ISessionFactoryImplementor factory)
 			return factory.ConnectionProvider.Driver is SqlClientDriver;
 		}
 
+		private bool _isSnapshotIsolationAlreadyAllowed;
+
+		private void CheckAllowSnapshotIsolation()
+		{
+			using (ISession session = OpenSession())
+			{
+				using (var command = session.Connection.CreateCommand())
+				{
+					command.CommandText = $@"select snapshot_isolation_state_desc from sys.databases 
+	where name = '{session.Connection.Database}'";
+					_isSnapshotIsolationAlreadyAllowed =
+						StringComparer.OrdinalIgnoreCase.Equals(command.ExecuteScalar() as string, "on");
+				}
+			}
+		}
+
 		private void SetAllowSnapshotIsolation(bool on)
 		{
 			using (ISession session = OpenSession())
 			{
```

Comment:
Can you remove curly braces from nested usings?

Reply:
Done.